rotkehlchen/chain/ethereum/oracles/uniswap.py

```diff
@@ -0,0 +1,385 @@
+                if pool != ZERO_ADDRESS:
+                    return [pool]
+
+        if from_asset == to_asset:
+            return []
+
+        # Try to find one asset that can be used between from_asset and to_asset
+        link_asset = None
+        found_first_link, found_second_link = False, False
+        for asset in self.routing_assets:
+            if asset != from_asset:
+                pools = self.get_pool(from_asset, asset)
+                for pool in pools:
+                    if pool != ZERO_ADDRESS:
+                        found_first_link = True
+                        output.append(pool)
+                        link_asset = asset
+                        break
+                if found_first_link:
+                    # If we have an asset lets see if we can pair with to_asset
+                    assert link_asset is not None
+                    pools = self.get_pool(link_asset, to_asset)
+                    for pool in pools:
+                        if pool != ZERO_ADDRESS:
+                            found_second_link = True
+                            output.append(pool)
+                            break
+                    if found_second_link:
+                        return output
+
+        # if we reach this point it means that we need 2 more jumps
+        if not found_first_link:
+            return []
+
+        second_link_asset = None
+        for asset in self.routing_assets:
+            if asset != to_asset:
+                pools = self.get_pool(asset, to_asset)
+                for pool in pools:
+                    if pool != ZERO_ADDRESS:
+                        found_second_link = True
+                        second_link_asset = asset
+                        output.append(pool)
+                        break
+                if found_second_link:
+                    break
+
+        if not found_second_link:
+            return []
+
+        assert second_link_asset is not None
+        assert link_asset is not None
+        pools = self.get_pool(link_asset, second_link_asset)
+        for pool in pools:
+            if pool != ZERO_ADDRESS:
+                output.insert(1, pool)
+                return output
+
+        return []
+
+    def get_price(
+        self,
+        from_asset: Asset,
+        to_asset: Asset,
+        block_identifier: BlockIdentifier,
+    ) -> Price:
+        """
+        Return the price of from_asset to to_asset at the block block_identifier.
+        External oracles are used if non eth tokens are used.
+
+        Can raise:
+        - PriceQueryUnsupportedAsset
+        - RemoteError
+        """
+        log.debug(f'Searching price for {from_asset} to {to_asset} at {block_identifier!r}')
+        if not (from_asset.is_eth_token() or to_asset.is_eth_token()):
+            raise PriceQueryUnsupportedAsset(
+                f'Neither {from_asset} nor {to_asset} are ethereum tokens for the uniswap oracle',
+            )
+
+        # Use weth internally
```

Comment:
Because Uniswap (both V2 and V3) use weth instead of eth in most of the pools. If you check the ETH tag in uniswap v3 you can see that is using the weth address https://info.uniswap.org/#/tokens/0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2 and same for v2 https://v2.info.uniswap.org/token/0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2

Reply:
Write that then in the comment. Make it fit in one line.